tools/cli/admin_task_list_commands.go

```diff
@@ -214,38 +254,56 @@ func AdminUpdateTaskListPartitionConfig(c *cli.Context) error {
 	}
 	tl := &types.TaskList{Name: taskList, Kind: types.TaskListKindNormal.Ptr()}
 	if !force {
-		err = validateChange(ctx, frontendClient, domain, tl, taskListType, cfg)
-		if err != nil {
-			return commoncli.Problem("Potentially unsafe operation. Specify '--force' to proceed anyway: ", err)
+		hasPollers := false
+		var errors []error
+		for _, tlType := range taskListTypes {
+			typeHasPollers, typeErr := validateChange(ctx, frontendClient, domain, tl, tlType.Ptr(), cfg)
+			if typeErr != nil {
+				errors = append(errors, fmt.Errorf("%s:%s failed validation: %w", tl.Name, tlType, typeErr))
+			}
+			hasPollers = typeHasPollers || hasPollers
+		}
+		if len(errors) > 0 {
+			return commoncli.Problem("Potentially unsafe operation. Specify '--force' to proceed anyway", multierr.Combine(errors...))
+		}
+		if !hasPollers {
```

Comment:
do we want to silently allow changing config when specifying more than one type, but only one has pollers?  kinda seems like a "warn first" scenario to me.

unless I'm misreading this?

Reply:
What you're describing is correct.

We pretty frequently partition a tasklist for both types despite only a single type being in use. Looking at our DynamicConfig we've never once partitioned a TaskList and used the TaskListType as one of the filters. So I don't think it provides better continuity and doesn't really create any problems for the system.

The intention behind this check is to catch cases where we specify the incorrect name. We recently had this happen with a large TaskList that we intended to change to 10 partitions but we partitioned `task_list_name` instead of `task-list-name` which caused some really bad hotsharding during the next monthly burst.

Reply:
yea, definitely useful either way.  and makes sense if it's kinda common 👍